modules/library/referencing/src/main/java/org/geotools/referencing/factory/DeferredAuthorityFactory.java

```diff
@@ -109,7 +110,7 @@ public boolean isAvailable() {
     @Override
     protected final AbstractAuthorityFactory getBackingStore() throws FactoryException {
         if (backingStore == null) {
-            synchronized (this) {
+            synchronized (ReferencingFactoryFinder.class) {
```

Comment:
I believe some javadoc should explain why the synchronization is done on another class rather than "this". The old code seems obvious, even if leading to a deadlock, this one should have an explanation.